services/brig/src/Brig/Team/DB.hs

```diff
@@ -174,6 +180,42 @@ lookupInvitationInfo ic@(InvitationCode c)
     cqlInvitationInfo :: PrepQuery R (Identity InvitationCode) (TeamId, InvitationId)
     cqlInvitationInfo = "SELECT team, id FROM team_invitation_info WHERE code = ?"
 
+lookupInvitationByEmail :: (Log.MonadLogger m, MonadClient m) => Email -> m (Maybe Invitation)
+lookupInvitationByEmail e = lookupInvitationEmail e >>= \case
+  Just InvitationInfo {..} -> lookupInvitation iiTeam iiInvId
+  _ -> return Nothing
+
+lookupInvitationEmail :: (Log.MonadLogger m, MonadClient m) => Email -> m (Maybe InvitationInfo)
+lookupInvitationEmail email = do
+  res <- retry x1 (query1 cqlInvitationEmail (params Quorum (Identity email)))
+  case res of
+    Nothing -> return Nothing
+    Just r -> toInvitationInfo r
+  where
+    toInvitationInfo (teamIds, invId, codes) = case (fromSet teamIds, fromSet codes) of
+      ([], []) -> do
+        -- unexpected, we (should) always insert a teamId
+        Log.warn $ Log.msg (Log.val "team_invidation_email: inconsistent database, row exists but empty teamId set")
+        return Nothing
+      (t : [], code : []) ->
+        -- one invite pending
+        return $ Just $ InvitationInfo code t invId
+      (t : [], code : _code2 : _) ->
+        -- multiple invites, but all from the same team, so can be deemed equivalent (TODO: can they? Different access role?)
+        return $ Just $ InvitationInfo code t invId
+      (_ : _ : _, _ : _ : _) -> do
+        -- edge case: more than one pending invite
+        -- We can treat the "none or multiple pending invites" the same way in the UI:
+        -- "please talk to your Wire administrator to figure out what to do"
+        Log.info $ Log.msg (Log.val "team_invidation_email: multiple pending invites from different team for the same email")
+        return $ Nothing
+      (_, _) -> do
```

Comment:
If you change the schema as I've described, I think you can get rid of this case.

Reply:
Indeed.